### PR TITLE
[DON' MERGE] Release 3.4.0

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/test/cache.config [hunk 1]
```diff
@@ -48,6 +48,7 @@ params {
     vep_cache_version = 110
     vep_genome        = 'WBcel235'
     vep_species       = 'caenorhabditis_elegans'
+    ngscheckmate_bed  = params.test_data['homo_sapiens']['genome']['ngscheckmate_bed']
```
Comment (maxulysse) [inline conf/test/cache.config:51]: I think we need the bcftools annotate files in this config file too

**Review by maxulysse — COMMENTED** [review]
Looking good

File: workflows/sarek.nf [hunk 2]
```diff
@@ -48,12 +48,18 @@ def checkPathParamList = [
     params.known_snps_tbi,
     params.mappability,
     params.multiqc_config,
+    params.ngscheckmate_bed,
     params.pon,
     params.pon_tbi,
+    params.sentieon_dnascope_model,
+    params.snpeff_cache,
```
Comment (maxulysse) [inline workflows/sarek.nf:55]:
```suggestion
```
I think we removed both vep_cache and snpeff_cache from there

File: workflows/sarek.nf [hunk 3]
```diff
@@ -48,12 +48,18 @@ def checkPathParamList = [
...
     params.spliceai_indel,
     params.spliceai_indel_tbi,
     params.spliceai_snv,
-    params.spliceai_snv_tbi
+    params.spliceai_snv_tbi,
+    params.bcftools_annotations,
+    params.bcftools_annotations_index,
+    params.bcftools_header_lines
```
Comment (maxulysse) [inline workflows/sarek.nf:62]: why?

File: subworkflows/local/samplesheet_to_channel/main.nf [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+        }
+        if (params.tools && (params.tools.split(',').contains('haplotypecaller') || params.tools.split(',').contains('sentieon_haplotyper') || params.tools.split(',').contains('sentieon_dnascope'))) {
+            log.warn "If GATK's Haplotypecaller, Sentieon's Dnascpe or Sentieon's Haplotyper is specified, without `--dbsnp` or `--known_indels no filtering will be done. For filtering, please provide at least one of `--dbsnp` or `--known_indels`.\nFor more information see FilterVariantTranches (single-sample, default): https://gatk.broadinstitute.org/hc/en-us/articles/5358928898971-FilterVariantTranches\nFor more information see VariantRecalibration (--joint_germline): https://gatk.broadinstitute.org/hc/en-us/articles/5358906115227-VariantRecalibrator\nFor more information on GATK Best practice germline variant calling: https://gatk.broadinstitute.org/hc/en-us/articles/360035535932-Germline-short-variant-discovery-SNPs-Indels-"
```
Comment (mirpedrol) [inline subworkflows/local/samplesheet_to_channel/main.nf:181]:
```suggestion
            log.warn "If GATK's Haplotypecaller, Sentieon's Dnascope or Sentieon's Haplotyper is specified, without `--dbsnp` or `--known_indels no filtering will be done. For filtering, please provide at least one of `--dbsnp` or `--known_indels`.\nFor more information see FilterVariantTranches (single-sample, default): https://gatk.broadinstitute.org/hc/en-us/articles/5358928898971-FilterVariantTranches\nFor more information see VariantRecalibration (--joint_germline): https://gatk.broadinstitute.org/hc/en-us/articles/5358906115227-VariantRecalibrator\nFor more information on GATK Best practice germline variant calling: https://gatk.broadinstitute.org/hc/en-us/articles/360035535932-Germline-short-variant-discovery-SNPs-Indels-"
```

File: workflows/sarek.nf [hunk 5]
```diff
@@ -481,6 +236,17 @@ include { MULTIQC                                     } from '../modules/nf-core
 
 workflow SAREK {
 
+	// Parse samplesheet
+	// Set input, can either be from --input or from automatic retrieval in WorkflowSarek.groovy
+	if (params.input) {
+        ch_from_samplesheet = params.build_only_index ? Channel.empty() : Channel.fromSamplesheet("input")
+	} else {
+        ch_from_samplesheet = params.build_only_index ? Channel.empty() : Channel.fromSamplesheet("input_restart")
+	}
```
Comment (maxulysse) [inline workflows/sarek.nf:245]:
```suggestion
	ch_from_samplesheet = params.build_only_index ? Channel.empty() : params.input ? Channel.fromSamplesheet("input") : Channel.fromSamplesheet("input_restart")
```

**Review by LaurenceKuhl — APPROVED** [review]

File: subworkflows/local/samplesheet_to_channel/main.nf [hunk 6]
```diff
@@ -0,0 +1,290 @@
+workflow  SAMPLESHEET_TO_CHANNEL{
+
+    take:
+    ch_from_samplesheet
+
+    main:
+    ch_from_samplesheet.dump(tag:"ch_from_samplesheet")
```
Comment (maxulysse) [inline subworkflows/local/samplesheet_to_channel/main.nf:7]: We should really have some more dump operator everywhere.

BTW, thanks @RaqManzano for that subworkflows, the code reads nicer in the main workflow thanks to that

**Review by mirpedrol — APPROVED** [review]
LGTM

**Comment by maxulysse** [comment]
PR should be from dev to master, not your branch

File: CHANGELOG.md [hunk 7]
```diff
@@ -38,6 +87,7 @@ A lake near the Rapaselet delta.
 ### Added
 
 - [#1231](https://github.com/nf-core/sarek/pull/1231) - Back to dev
+- [#1244](https://github.com/nf-core/sarek/pull/1244) - Add bcf annotate module
```
[the remark this answers is not in the document]
Comment (maxulysse) [reply CHANGELOG.md:90]: good point

File: CHANGELOG.md [hunk 8]
```diff
@@ -72,6 +122,7 @@ Rapaselet is a delta formed by the Rapaätno river between the Bielloriehppe mas
 - [#1173](https://github.com/nf-core/sarek/pull/1173) - CI tests for VQSR track with stub runs
 - [#1122](https://github.com/nf-core/sarek/pull/1122), [#1196](https://github.com/nf-core/sarek/pull/1196) - Add `annotation cache` functionality
 - [#1184](https://github.com/nf-core/sarek/pull/1184) - Stub-based CI-test of Sentieon joint-germline variant-calling with VQSR
+- [#1288](https://github.com/nf-core/sarek/pull/1288) - Add nf-test continuous integration.
```
[the remark this answers is not in the document]
Comment (maxulysse) [reply CHANGELOG.md:125]: Yes

**Comment by maxulysse** [comment]
> PR should be from dev to master, not your branch

I think it'll be good to have the sentieon tests for the release

**Comment by FriederikeHanssen** [comment]
> PR should be from dev to master, not your branch

🤦‍♀️ will open a new PR, but keep this one since people started reviewing it.

**Comment by FriederikeHanssen** [comment]
Reviews addressed in #1324 . Will close this PR and merge reviews into dev. This PR is superseded by #1321

File: workflows/sarek.nf (hunk 3 above)
Comment (maxulysse) [reply workflows/sarek.nf:62]: My question was more why not sorted actually

Comment (FriederikeHanssen) [reply workflows/sarek.nf:62]: should've known

Comment (FriederikeHanssen) [reply workflows/sarek.nf:62]: fixed

**Pull request closed without merging**